packages/sdk/src/credential/teamsUserCredential.browser.ts

```diff
@@ -178,21 +184,63 @@ export class TeamsUserCredential implements TokenCredential {
       return ssoToken;
     } else {
       internalLogger.info("Get access token with scopes: " + scopeStr);
-      const cachedKey = await this.getAccessTokenCacheKey(scopeStr);
-      const cachedToken = this.getTokenCache(cachedKey);
-
-      if (cachedToken) {
-        if (!this.isAccessTokenNearExpired(cachedToken)) {
-          internalLogger.verbose("Get access token from cache");
-          return cachedToken;
-        } else {
-          internalLogger.verbose("Cached access token is expired");
+
+      if (!this.initialized) {
+        await this.init();
+      }
+
+      // Get domain from login start page. Will be used in redirect uri.
+      let domain;
+      try {
+        domain = new URL(this.config.initiateLoginEndpoint!).origin;
+      } catch (error: any) {
+        const failedToParseLoginEndpoint = `Failed to parse INITIATE_LOGIN_ENDPOINT in config. Reson: ${error.message}`;
+        internalLogger.error(failedToParseLoginEndpoint);
+        throw new ErrorWithCode(failedToParseLoginEndpoint, ErrorCode.InternalError);
+      }
+
+      let tokenResponse;
+      let errorMessage = "";
+      const scopesArray = typeof scopes === "string" ? scopes.split(" ") : scopes;
+
+      // First try to get Access Token from cache.
+      try {
+        const account = this.msalInstance!.getAccountByUsername(this.loginHint!);
+        const scopesRequestForAcquireTokenSilent = {
+          scopes: scopesArray,
+          account: account ?? undefined,
+          redirectUri: `${domain}/blank-auth-end.html`,
```

Comment:
if we use this redirectUri, does it mean the cache set in auth-end.html will not be used? as the redirectUri is different